app/models/user.rb

```diff
@@ -105,18 +105,49 @@ def token
     JsonWebToken.encode(user_id: self.id, roles: [], email: self.email, verified: verified)
   end
 
+  # For social media logins, When given a user's info creates their database entry if they do not already have one, and
+  # sets the redirect path for the frontend to go to after the user is logged in
+  # for the first time.
+  #
+  # Requires that all data contains :first_name, :last_name, :email, :zip, and :password keys.
+  # For example: { user: {first_name: "Jane"},{last_name: "Doe"},{ email: "jane@example.com" },{ zip: "12345"},{ password: "Th!sIs@Pa22w0rd"} }
```

Comment:
This is an invalid structure, with all of these hashes separated by commas.  Can you pls update.

Reply:
I took out the commas, let me know if what it is now is what you're looking for in terms of format.